### ISSUE TITLE
GIT: Large empty space between the file path and the Show Actions button

### ISSUE THREAD

**Issue by troizet** [apache/netbeans#5680]
### Apache NetBeans version

Apache NetBeans 17

### What happened

![Снимок экрана от 2023-03-19 19-10-09(2)](https://user-images.githubusercontent.com/9607501/226174927-20a18024-bd6c-4078-b99b-d50d9652402c.png)

If you click on `Show all files`:
![Снимок экрана от 2023-03-19 19-11-12](https://user-images.githubusercontent.com/9607501/226174992-69a8af6b-fa59-450e-9306-e2024c77544a.png)

On the previous commit:
![Снимок экрана от 2023-03-19 19-11-58](https://user-images.githubusercontent.com/9607501/226175021-ca419a61-81b3-4aa4-b1bd-e281b54c2e01.png)

![Снимок экрана от 2023-03-19 19-12-26](https://user-images.githubusercontent.com/9607501/226175030-e7dcbf25-c50e-4360-8fa2-3d67b9c6a78a.png)

### How to reproduce

 - Open the `php.editor` module
 - `Git -> Show history` for file `php/php.editor/src/org/netbeans/modules/php/editor/verification/ReturnTypeHintError.java`
 - In summary find commit `f55ac21` and expand for a list of files

### Did this work correctly in an earlier version?

No / Don't know

### Operating System

Ubuntu 22.04.2 LTS

### JDK

OpenJDK Runtime Environment 11.0.18+10-post-Ubuntu-0ubuntu122.04

### Apache NetBeans packaging

Apache NetBeans binary zip

### Are you willing to submit a pull request?

No

**Comment by mbien** [comment]
this seems to be an optimization:
https://github.com/apache/netbeans/blob/67cc93027d9ebbb69bf5e00fd6e529f9bf960a4b/ide/versioning.util/src/org/netbeans/modules/versioning/history/SummaryCellRenderer.java#L203

it computes the line size only if the commit has less than 20 "events". An event is essentially a change to a file, like add/remove/modify.

We might be able to remove this check, or bump it to a larger number.

**Comment by mbien** [comment]
-> https://github.com/apache/netbeans/pull/7128 PR should create a dev-build to play around with

**Comment by troizet** [comment]
@mbien 
I don't seem to have described the problem clearly, sorry.

The location of the Show Actions button depends on the longest file path in the commit, even if that path is not displayed. 
And when the commit file list is minimized to a single file, it looks strange.  And sometimes because of one file with a long path, the button can go off screen for all the other files in the commit.
I think it would be better to remove the empty space between the file path and the button.

**Comment by mbien** [comment]
@troizet in your screenshot it shows only one file for some reason which makes it look weird if it happens to be the shortest path.

I can't explain why you see only one file. I also question a bit how useful the "show all files" button is. I am thinking about making this the default and removing the button, so that when a user expands a commit, there would be always all files.

For me personally, I think it looks a bit better when its formatted like a table, yes there can be the odd situation when a very short path is next to a very long path - but this is already mitigated a little bit by limiting the max empty space to 400 chars.

**Comment by mbien** [comment]
ah! thanks, that was the missing piece - now I understand how to get there.

I check atm if I can implement a file filter with highlighting, after that I check if this can be improved somehow.

**Comment by troizet** [comment]
> I also question a bit how useful the "show all files" button is.
I think it would be fine, but just not for the commit history for a particular file.

**Comment by mbien** [comment]
this should work now I hope. PR is updated (see dev-build).

**Comment by troizet** [comment]
Tried dev-build. Everything works fine. Thank you very much!

**Comment by mbien** [comment]
@troizet wow that was quick! Thanks for checking.
